src/docs/testing/best-practices.md

```diff
@@ -0,0 +1,173 @@
+---
+title: Performance best practices
+short-title: Best practices
+description: How to ensure that your Flutter app is performant.
+---
+
+Generally, Flutter applications are performant by default,
+so you only need to avoid common pitfalls to get excellent performance
+instead of needing to micro-optimize with complicated profiling tools.
+These best recommendations will help you write the most performant
+Flutter app possible.
+
+## Best practices
+
+How do you design a Flutter app to most efficiently render your scenes?
+In particular, how do you ensure that the painting code generated by the
+framework is as efficient as possible? Here are a few things to consider
+when designing your app:
+
+### Call setState thoughtfully
+
+* Avoid repetitive and costly work in `build()` methods since `build()`
+  can be invoked frequently when ancestor Widgets rebuild.
+* Localize the `setState()` call to the part of the subtree whose UI
+  actually needs to change. Avoid calling calling setState() high up in
+  the tree if the change is contained to a small part of the tree.
+* Re-use the same child widget instance when building stops that
+  child subtree from re-building when the parent rebuilds.
+  [PENDING: rephrase]
```

Comment:
Is this going to be rephrased before the page goes out?

Reply:
Good catch!